### PR TITLE
Add AgencyStopPurposeGroupView and AgencyStopGroupByPurposeView 

### PR DESCRIPTION
**Pull request by copelco**
- Add `AgencyStopPurposeGroupView` and `AgencyStopGroupByPurposeView` views to calculate stops by their respective stop purpose
- Add `react-chartsjs-2` new graphing library to simplify frontend.
- Add news graphs for traffic stops

Preview:
Desktop:
Screen Shot 2023-04-19 at 5 40 21 PM
Screen Shot 2023-05-16 at 12 07 32 PM

### CODE REVIEW

**Comment by Afani97** [comment]
@copelco Can you change the base branch to `develop`? I think this can be separate from the range picker branch.

File: frontend/src/Components/Elements/Table/TableModal.js [hunk 1]
```diff
@@ -499,7 +500,7 @@ function TableModal({ chartState, dataSet, columns, isOpen, closeModal }) {
   };
 
   const closeRangePicker = () => {
-    setShowDateRagePicker(false);
+    setShowDateRangePicker(false);
```
Comment (Afani97) [inline frontend/src/Components/Elements/Table/TableModal.js:503]: Fix typo

File: frontend/src/Components/Charts/TrafficStops/TrafficStops.js [hunk 2]
```diff
@@ -81,6 +83,18 @@ function TrafficStops(props) {
   const renderMetaTags = useMetaTags();
   const [renderTableModal, { openModal }] = useTableModal();
 
+  const [stopPurposeGroupsData, setStopPurposeGroups] = useState({ labels: [], datasets: [] });
+
+  // Build Stop Purpose Groups
+  useEffect(() => {
+    axios
+      .get(`/api/agency/${agencyId}/stop-purpose-groups/`)
+      .then((res) => {
```
Comment (Afani97) [inline frontend/src/Components/Charts/TrafficStops/TrafficStops.js:92]: So much simpler to reason where the data comes from.

File: nc/views.py [hunk 3]
```diff
@@ -338,3 +364,130 @@ def post(self, request):
...
+        black_mask = df[purpose]["driver_race_comb"] == "Black"
+        hispanic_mask = df[purpose]["driver_race_comb"] == "Hispanic"
+        asian_mask = df[purpose]["driver_race_comb"] == "Asian"
+        native_american_mask = df[purpose]["driver_race_comb"] == "Native American"
+        other_mask = df[purpose]["driver_race_comb"] == "Other"
+        return {
+            "labels": df.year.unique(),
+            "datasets": [
+                {
+                    "label": "White",
```
Comment (Afani97) [inline nc/views.py:420]: @copelco Is there simpler way of doing this?

File: nc/models.py [hunk 4]
```diff
@@ -4,6 +4,52 @@
...
+    SPEED_LIMIT_VIOLATION = 1, "Speed Limit Violation"  # Safety Violation
+    STOP_LIGHT_SIGN_VIOLATION = 2, "Stop Light/Sign Violation"  # Safety Violation
+    DRIVING_WHILE_IMPAIRED = 3, "Driving While Impaired"  # Safety Violation
+    SAFE_MOVEMENT_VIOLATION = 4, "Safe Movement Violation"  # Safety Violation
+    VEHICLE_EQUIPMENT_VIOLATION = 5, "Vehicle Equipment Violation"  # Regulatory and Equipment
+    VEHICLE_REGULATORY_VIOLATION = 6, "Vehicle Regulatory Violation"  # Regulatory and Equipment
+    OTHER_MOTOR_VEHICLE_VIOLATION = 9, "Other Motor Vehicle Violation"  # Regulatory and Equipment
+    SEAT_BELT_VIOLATION = 7, "Seat Belt Violation"  # Investigatory
+    INVESTIGATION = 8, "Investigation"  # Investigatory
+    CHECKPOINT = 10, "Checkpoint"  # Investigatory
```
Comment (copelco) [inline nc/models.py:18]:
```suggestion
    SEAT_BELT_VIOLATION = 7, "Seat Belt Violation"  # Regulatory and Equipment
    INVESTIGATION = 8, "Investigation"  # Other
    CHECKPOINT = 10, "Checkpoint"  # Other
```

**Review by copelco — COMMENTED** [review]
Looks good to me. I can't approve it since I open the PR, but 👍

**Pull request merged**